cassovary-core/src/main/scala/com/twitter/cassovary/util/io/MemoryMappedIntLongSource.scala

```diff
@@ -0,0 +1,48 @@
+package com.twitter.cassovary.util.io
+
+import java.io.File
+import java.nio.MappedByteBuffer
+import java.nio.channels.FileChannel
+import java.nio.channels.FileChannel.MapMode
+import java.nio.file.StandardOpenOption
+
+/**
+ * Represents an arbitrarily large array of bytes which can be interpreted as ints or longs.
+ */
+trait IntLongSource {
+  /** Returns the int at the given byte index, which must be a multiple of 4. */
+  def getInt(index: Long): Int
+  /** Returns the Long at the given byte index, which must be a multiple of 8. */
+  def getLong(index: Long): Long
+}
+
+/**
+ * Wraps a sequence of FileChannels to enable random access on a memory mapped file of arbitrary size.
+ * Motivation: FileChannel.open only supports 2GB at a time.
+ * @param file The file containing binary data.
+ */
+
+class MemoryMappedIntLongSource(file: File) extends IntLongSource {
+  def this(fileName: String) = this(new File(fileName))
+
+  private val fileChannel = FileChannel.open(file.toPath, StandardOpenOption.READ)
+
+  // Each buffer uses int addressing, so can only access 2GB.  We'll use multiple buffers,
+  // and access 2^30 bytes per buffer
```

Comment:
Actually, I just tried 2^31, and FileChannel requires size at most Integer.MAX_VALUE and throws an exception given size 2^31, so we can't actually have a buffer of size 2^31.  This is a strange API decision in Java.  I'll stick with 2^30.